### PR TITLE
feat(stackdriver): add support for gce_instance MR

### PR DESCRIPTION
**Pull request by douglas-reid**
**What this PR does / why we need it**:
Adds support for `gce_instance` MR type to the Stackdriver extension code.

This PR also updates the metadata keys to match the keys in https://github.com/istio/istio/blob/master/pkg/bootstrap/platform/gcp.go#L27-L32.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

See: istio/istio#17081

### CODE REVIEW

File: extensions/stackdriver/common/constants.h [hunk 1]
```diff
@@ -45,17 +45,21 @@ constexpr char kIstioMetricPrefix[] = "istio.io/service/";
...
 constexpr char kNamespaceNameLabel[] = "namespace_name";
 constexpr char kPodNameLabel[] = "pod_name";
 constexpr char kContainerNameLabel[] = "container_name";
+constexpr char kGCEInstanceIDLabel[] = "instance_id";
+constexpr char kZoneLabel[] = "zone";
 
 // GCP node metadata key
-constexpr char kGCPClusterLocationKey[] = "gcp_cluster_location";
-constexpr char kGCPClusterNameKey[] = "gcp_cluster_name";
+constexpr char kGCPClusterLocationKey[] = "gcp_location";
```
Comment (bianpengyuan) [inline extensions/stackdriver/common/constants.h]: Rename this to `kGCPLocationKey`?

Comment (douglas-reid) [reply extensions/stackdriver/common/constants.h]: Good idea. Done.

File: extensions/stackdriver/log/logger.cc [hunk 2]
```diff
@@ -48,10 +49,18 @@ Logger::Logger(const ::wasm::common::NodeInfo& local_node_info,
   log_entries_request_->set_log_name("projects/" + project_id + "/logs/" +
                                      kServerAccessLogName);
 
+  std::string resource_type = Common::kContainerMonitoredResource;
```
Comment (bianpengyuan) [inline extensions/stackdriver/log/logger.cc]: nit: s/std::string/auto&. Save a copy of string.

Comment (douglas-reid) [reply extensions/stackdriver/log/logger.cc]: done.

**Pull request merged**